src/pybind/mgr/nfs/cluster.py

```diff
@@ -126,16 +144,29 @@ def _show_nfs_cluster_info(self, cluster_id):
                 ('2404:6800:4009:80d::200e', 2049, 0, 0))]
                 """
                 try:
-                    host_ip.append({
+                    backends.append({
                             "hostname": cluster.hostname,
-                            "ip": list(set([ip[4][0] for ip in socket.getaddrinfo(
-                                cluster.hostname, 2049, flags=socket.AI_CANONNAME,
-                                type=socket.SOCK_STREAM)])),
-                            "port": 2049  # Default ganesha port
+                            "ip": cluster.ip or resolve_ip(cluster.hostname),
+                            "port": cluster.ports[0]
                             })
                 except socket.gaierror:
```

Comment:
Change this exception to `OrchestratorError`

https://github.com/ceph/ceph/blob/f8e872ce2fc7d6f0f960d51942e3e9cc4d8cc265/src/pybind/mgr/cephadm/utils.py#L107-L117